cns/restserver/api.go

```diff
@@ -1522,3 +1522,43 @@ func (service *HTTPRestService) nmAgentSupportedApisHandler(w http.ResponseWrite
 
 	logger.Response(service.Name, nmAgentSupportedApisResponse, resp.ReturnCode, serviceErr)
 }
+
+// getVMUniqueID retrieves VMUniqueID from the IMDS
+func (service *HTTPRestService) getVMUniqueID(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getVMUniqueID")
+	logger.Request(service.Name, "getVMUniqueID", nil)
+	ctx := r.Context()
+
+	switch r.Method {
+	case http.MethodGet:
+		vmUniqueID, err := service.imdsClient.GetVMUniqueID(ctx)
+		var resp cns.GetVMUniqueIDResponse
+		httpStatus := http.StatusOK
+		if err != nil {
+			resp = cns.GetVMUniqueIDResponse{
+				Response: cns.Response{
+					ReturnCode: types.UnexpectedError,
+					Message:    errors.Wrap(err, "failed to get vmuniqueid").Error(),
+				},
+			}
+			httpStatus = http.StatusInternalServerError
+		} else {
+			resp = cns.GetVMUniqueIDResponse{
+				Response: cns.Response{
+					ReturnCode: types.Success,
+				},
+				VMUniqueID: vmUniqueID,
+			}
+		}
```

Comment:
I think style-wise, we'd prefer to handle errors right away and return early. I'd consider refactoring to something like
```golang
vmUniqueID, err := service.imdsClient.GetVMUniqueID(ctx)
if err != nil {
    resp := cns.GetVMUniqueIDResponse{
        Response: cns.Response{
            ReturnCode: types.UnexpectedError,
            Message: errors.Wrap(err, "failed to get vm unique ID").Error(),
        },
    }
    logger.Response(service.Name, resp, resp.Response.ReturnCode, err)
    respondJSON(w, http.StatusInternalServerError, resp)
    return
}

resp := cns.GetVMUniqueIDResponse{
    Response: cns.Response{
        ReturnCode: types.Success,
    },
    VMUniqueID: vmUniqueID,
}

....
```

Reply:
I have fixed it. Before idea was to remove some duplicate code.